errors/format.go

```diff
@@ -0,0 +1,67 @@
+// Copyright © 2017 The Things Network
+// Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+
+package errors
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/gotnospirit/messageformat"
```

Comment:
I still think `Error` should be a concrete type, which has `MarshalJSON` etc. defined and functions should use small interfaces (`JSONMarshaller` etc.) to implement functionality, but ok